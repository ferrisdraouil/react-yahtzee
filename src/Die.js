import React, { Component } from 'react';
import './Die.css';

class Die extends Component {
  render() {
    return (
      <button
        className={this.props.locked ? 'Die Die-locked' : 'Die'}
        onClick={() => this.props.handleClick(this.props.idx)}
      >
        // can not be clicked if the this.props.locked is true
        {this.props.val}
      </button>
    );
  }
}

export default Die;
