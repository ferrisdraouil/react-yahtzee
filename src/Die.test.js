import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Die from './Die';

it('renders without crashing', function() {
  shallow(<Die />);
});

it('matches unlocked snapshot', function() {
  let wrapper = shallow(<Die locked={false} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it('matches locked snapshot', function() {
  let wrapper = shallow(<Die locked={true} />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
