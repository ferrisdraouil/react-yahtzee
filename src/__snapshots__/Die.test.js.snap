// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches locked snapshot 1`] = `
<button
  className="Die Die-locked"
  onClick={[Function]}
>
  // can not be clicked if the this.props.locked is true
</button>
`;

exports[`matches unlocked snapshot 1`] = `
<button
  className="Die"
  onClick={[Function]}
>
  // can not be clicked if the this.props.locked is true
</button>
`;
